lib/ui/painting/image_decoding.cc

```diff
@@ -35,20 +35,20 @@ sk_sp<SkImage> DecodeImage(std::vector<uint8_t> buffer) {
   if (sk_data == nullptr)
     return nullptr;
 
-  std::unique_ptr<SkImageGenerator> generator(
-      SkImageGenerator::NewFromEncoded(sk_data.get()));
+  sk_sp<SkImage> image = SkImage::MakeFromEncoded(std::move(sk_data));
```

Comment:
In case the image is returned without uploading to the GPU directly (in case `ResourceContext::Get()` is `nullptr`), SkImage would take ownership of the `SkData` correct? In that case, the backing of the SkData would not survive this stack frame. We should probably use `SkData::MakeWithCopy`. In the previous code path, the SkImageGenerator did not take the `SkData` instance.